test/functional/setup.js

```diff
@@ -181,6 +183,12 @@ before(function () {
             return openRemoteBrowsers();
         })
         .then(() => {
+            if (config.useLocalBrowsers && !config.useHeadlessBrowsers && os.linux && !detectDisplay())
```

Comment:
I'm afraid we can't do this thing, since even if your test doesn't require the browser specified in the task configuration, it starts together with all other functional tests that rely on that browser.

Reply:
Shame on me. I didn't notice that the functional tests (in local and non-headless browsers) are running in Travis with enabled `xvfb` service.